superset/common/query_context_factory.py

```diff
@@ -128,7 +128,10 @@ def _apply_granularity(
 
         if granularity := query_object.granularity:
             filter_to_remove = None
-            if x_axis and x_axis in temporal_columns:
+            x_axis_column_name = (
+                x_axis["sqlExpression"] if isinstance(x_axis, dict) else x_axis
+            )
```

Comment:
We have a helper util for this called `get_column_name`, let's use that here.

Reply:
+1